modules/adhashBidAdapter.js

```diff
@@ -0,0 +1,90 @@
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { BANNER } from '../src/mediaTypes.js';
+
+export const spec = {
+  code: 'adhash',
+  url: 'https://bidder.adhash.org/rtb?version=1.0&prebid=true',
+  supportedMediaTypes: [ BANNER ],
+
+  isBidRequestValid: (bid) => {
+    try {
+      const { publisherId, platformURL } = bid.params;
+      return (
+        typeof publisherId === 'string' &&
+        publisherId.length === 42 &&
+        typeof platformURL === 'string' &&
+        platformURL.length >= 13
+      );
+    } catch (error) {
+      return false;
+    }
+  },
+
+  buildRequests: (validBidRequests, _bidderRequest) => {
+    const { url } = spec;
+    const bidRequests = [];
+    for (var i = 0; i < validBidRequests.length; i++) {
+      bidRequests.push({
+        method: 'POST',
+        url: url,
+        bidRequest: validBidRequests[i],
+        data: {
+          timezone: new Date().getTimezoneOffset() / 60,
+          referrer: document.referrer,
+          location: window.location.href,
+          publisherId: validBidRequests[i].params.publisherId,
+          size: {
+            screenWidth: window.screen.width,
+            screenHeight: window.screen.height
+          },
+          navigator: {
+            platform: window.navigator.platform,
+            language: window.navigator.language,
+            userAgent: window.navigator.userAgent
+          },
+          creatives: [{
+            size: validBidRequests[i].sizes[0].join('x'),
+            position: validBidRequests[i].adUnitCode
+          }],
+          blockedCreatives: [],
+          currentTimestamp: new Date().getTime(),
+          recentAds: []
+        },
+        options: {
+          withCredentials: false,
+          crossOrigin: true
+        },
+      });
+    }
+    return bidRequests;
```

Comment:
I noticed you're not passing GDPR consent, please update to include, it's not required but considering your adapter is returning EUR currency in your responses it may be useful to have.

Reply:
Thanks. I'm passing this as a parameter now.